In [1]:
import numpy as np
import pandas as pd
import seaborn as sn
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings('ignore')

In [2]:
accident = pd.read_csv('C:/Users/LAB4-StudentPC27/Desktop/ITD-3C_ICAL/datasets/accident_data.csv')

In [3]:
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [4]:
accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [5]:
accident.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [6]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

<h1>Data Types</h1>

In [7]:
accident.dtypes

Index                       object
Accident_Severity           object
Accident Date               object
Latitude                   float64
Light_Conditions            object
District Area               object
Longitude                  float64
Number_of_Casualties         int64
Number_of_Vehicles           int64
Road_Surface_Conditions     object
Road_Type                   object
Urban_or_Rural_Area         object
Weather_Conditions          object
Vehicle_Type                object
dtype: object

In [8]:
accident['Accident_Severity'].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

In [9]:
accident['Light_Conditions'].value_counts()

Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

In [10]:
accident['Urban_or_Rural_Area'].value_counts()

Urban_or_Rural_Area
Urban          421663
Rural          238990
Unallocated        11
Name: count, dtype: int64

In [11]:
accident['Road_Surface_Conditions'].value_counts()

Road_Surface_Conditions
Dry                     447821
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64

In [12]:
accident['Accident Date'].value_counts()

Accident Date
30-11-2019    704
31-01-2019    697
13-07-2019    692
13-11-2021    692
14-08-2019    688
             ... 
30-12-2022    171
25-12-2019    157
25-12-2022    145
10/1/2022     123
25-12-2020    118
Name: count, Length: 1461, dtype: int64

In [13]:
accident['Accident Date']= pd.to_datetime(accident['Accident Date'], dayfirst = True, errors = 'coerce')

In [14]:
accident.dtypes

Index                              object
Accident_Severity                  object
Accident Date              datetime64[ns]
Latitude                          float64
Light_Conditions                   object
District Area                      object
Longitude                         float64
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions            object
Road_Type                          object
Urban_or_Rural_Area                object
Weather_Conditions                 object
Vehicle_Type                       object
dtype: object

In [15]:
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
1       200701BS65737           Serious    2019-07-02  51.495029   
2       200701BS66127           Serious           NaT  51.517715   
3       200701BS66128           Serious           NaT  51.495478   
4       200701BS66837            Slight    2019-09-03  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight           NaT  57.374005   
660675  201091NM01881            Slight           NaT  57.232273   
660676  201091NM01935            Slight           NaT  57.585044   
660677  201091NM01964           Serious           NaT  57.214898   
660678  201091NM02142           Serious           NaT  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

<h1>Insight no. 1</h1>

<h1>In what area has the most serious accident in car happen 2019 - 2022</h1>

In [16]:
accident_car = accident[(accident['Vehicle_Type'] == 'Car') & (accident['Accident_Severity'] == 'Serious')]

In [17]:
accident_car

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
1       200701BS65737           Serious    2019-07-02  51.495029   
5       200701BS67159           Serious           NaT  51.497750   
16      200701BS67969           Serious    2019-10-01  51.493490   
17      200701BS68160           Serious    2019-10-10  51.508445   
...               ...               ...           ...        ...   
660619  201091NK04839           Serious           NaT  58.977273   
660629  201091NL00686           Serious    2022-02-11  57.687933   
660645  201091NL05842           Serious           NaT  60.162660   
660672  201091NM01085           Serious           NaT  57.170725   
660678  201091NM02142           Serious           NaT  57.575210   

             Light_Conditions           District Area  Longitude  \
0       Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                    Daylight  Kensington and Chelsea  -0.173647   
5                    Daylight  Kensington and Chelsea  -0.192556   
16                   Daylight  Kensington and Chelsea  -0.190420   
17                   Daylight  Kensington and Chelsea  -0.215622   
...                       ...                     ...        ...   
660619               Daylight          Orkney Islands  -2.963443   
660629               Daylight        Shetland Islands  -4.176860   
660645               Daylight        Shetland Islands  -1.165232   
660672               Daylight                Highland  -3.852421   
660678               Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
5                          2                   3                     Dry   
16                         1                   2                     Dry   
17                         1                   1                     Dry   
...                      ...                 ...                     ...   
660619                     1                   2                     Dry   
660629                     1                   2                     Dry   
660645                     1                   1                     Dry   
660672                     1                   2                    Snow   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
5       Single carriageway               Urban     Fine no high winds   
16        Dual carriageway               Urban     Fine no high winds   
17      Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
660619  Single carriageway               Rural     Fine no high winds   
660629  Single carriageway               Rural      Fine + high winds   
660645  Single carriageway               Rural      Fine + high winds   
660672  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

       Vehicle_Type  
0               Car  
1               Car  
5               Car  
16              Car  
17              Car  
...             ...  
660619          Car  
660629          Car  
660645          Car  
660672          Car  
660678          Car  

[66461 rows x 14 columns]

<h1>Theres is a total of 57% serious car accident happen in Urban Area and 42% serious car accident happen in Rural Area, means Urban area has the most accident rate happen to 2019 - 2022</h1>

In [18]:
accident_car['Urban_or_Rural_Area'].value_counts()

Urban_or_Rural_Area
Urban    37965
Rural    28494
Name: count, dtype: int64

In [19]:
accident_car['Urban_or_Rural_Area'].value_counts().sum()

np.int64(66459)

<h1>Insight no.2</h1>

<h1>What Road surface condition has the most critical serious accident happen in Car vehicle</h1>

In [20]:
accident_casualties = accident[(accident_car['Light_Conditions'] == 'Daylight') & (accident['Accident_Severity'] == 'Serious')]

In [21]:
accident_casualties

Index Accident_Severity Accident Date   Latitude  \
1       200701BS65737           Serious    2019-07-02  51.495029   
5       200701BS67159           Serious           NaT  51.497750   
16      200701BS67969           Serious    2019-10-01  51.493490   
17      200701BS68160           Serious    2019-10-10  51.508445   
19      200701BS68237           Serious           NaT  51.482988   
...               ...               ...           ...        ...   
660619  201091NK04839           Serious           NaT  58.977273   
660629  201091NL00686           Serious    2022-02-11  57.687933   
660645  201091NL05842           Serious           NaT  60.162660   
660672  201091NM01085           Serious           NaT  57.170725   
660678  201091NM02142           Serious           NaT  57.575210   

       Light_Conditions           District Area  Longitude  \
1              Daylight  Kensington and Chelsea  -0.173647   
5              Daylight  Kensington and Chelsea  -0.192556   
16             Daylight  Kensington and Chelsea  -0.190420   
17             Daylight  Kensington and Chelsea  -0.215622   
19             Daylight  Kensington and Chelsea  -0.174272   
...                 ...                     ...        ...   
660619         Daylight          Orkney Islands  -2.963443   
660629         Daylight        Shetland Islands  -4.176860   
660645         Daylight        Shetland Islands  -1.165232   
660672         Daylight                Highland  -3.852421   
660678         Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
1                          1                   2             Wet or damp   
5                          2                   3                     Dry   
16                         1                   2                     Dry   
17                         1                   1                     Dry   
19                         1                   4                     Dry   
...                      ...                 ...                     ...   
660619                     1                   2                     Dry   
660629                     1                   2                     Dry   
660645                     1                   1                     Dry   
660672                     1                   2                    Snow   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
1       Single carriageway               Urban  Raining no high winds   
5       Single carriageway               Urban     Fine no high winds   
16        Dual carriageway               Urban     Fine no high winds   
17      Single carriageway               Urban     Fine no high winds   
19      Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
660619  Single carriageway               Rural     Fine no high winds   
660629  Single carriageway               Rural      Fine + high winds   
660645  Single carriageway               Rural      Fine + high winds   
660672  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

       Vehicle_Type  
1               Car  
5               Car  
16              Car  
17              Car  
19              Car  
...             ...  
660619          Car  
660629          Car  
660645          Car  
660672          Car  
660678          Car  

[45757 rows x 14 columns]

<h1>The most serious car accident happen is in the Dry Surface with the total of 76% which means that the accident rate most happen in Dry road surface
</h1>

In [22]:
accident_casualties['Road_Surface_Conditions'].value_counts().sum()

np.int64(45730)

In [23]:
accident_casualties['Road_Surface_Conditions'].value_counts()

Road_Surface_Conditions
Dry                     35090
Wet or damp              9612
Frost or ice              745
Snow                      229
Flood over 3cm. deep       54
Name: count, dtype: int64

<h1>Insight no.3</h1>

<h1>How Many Serious accident happen in Daylight </h1>

In [24]:
accident[(accident['Accident_Severity'] == 'Serious') & (accident['Light_Conditions'] == 'Daylight')].count()

Index                      60759
Accident_Severity          60759
Accident Date              24453
Latitude                   60755
Light_Conditions           60759
District Area              60759
Longitude                  60755
Number_of_Casualties       60759
Number_of_Vehicles         60759
Road_Surface_Conditions    60720
Road_Type                  60410
Urban_or_Rural_Area        60758
Weather_Conditions         59972
Vehicle_Type               60759
dtype: int64

In [25]:
accident['Year'] = accident['Accident Date'].dt.year
accident['Month'] = accident['Accident Date'].dt.year
accident['Day'] = accident['Accident Date'].dt.year
accident['DayofWeek'] = accident['Accident Date'].dt.year

<h1>Insight no.4</h1>

In [26]:
df_year20 = accident[accident['Year'] == 2020]
df_year19 = accident[accident['Year'] == 2019]
df_year21 = accident[accident['Year'] == 2021]
df_year22 = accident[accident['Year'] == 2022]

In [27]:
df_year20

Index Accident_Severity Accident Date   Latitude  \
182120  200801BS70001           Serious    2020-01-02  51.491521   
182121  200801BS70002            Slight    2020-01-06  51.484909   
182122  200801BS70003           Serious    2020-01-12  51.509965   
182123  200801BS70004            Slight    2020-01-05  51.490631   
182124  200801BS70005            Slight    2020-01-03  51.509712   
...               ...               ...           ...        ...   
352694       2.01E+12            Slight    2020-11-01  55.267100   
352695       2.01E+12            Slight    2020-11-02  54.985218   
352696       2.01E+12            Slight    2020-11-06  55.126057   
352697       2.01E+12           Serious    2020-11-04  55.243971   
352702       2.01E+12            Slight    2020-12-06  54.983911   

              Light_Conditions           District Area  Longitude  \
182120   Darkness - lights lit  Kensington and Chelsea  -0.179406   
182121   Darkness - lights lit  Kensington and Chelsea  -0.176356   
182122                Daylight  Kensington and Chelsea  -0.197405   
182123   Darkness - lights lit  Kensington and Chelsea  -0.185779   
182124   Darkness - lights lit  Kensington and Chelsea  -0.192659   
...                        ...                     ...        ...   
352694  Darkness - no lighting   Dumfries and Galloway  -2.959324   
352695                Daylight   Dumfries and Galloway  -3.278743   
352696  Darkness - no lighting   Dumfries and Galloway  -3.344404   
352697  Darkness - no lighting   Dumfries and Galloway  -3.461163   
352702                Daylight   Dumfries and Galloway  -3.195094   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
182120                     1                   2                     Dry   
182121                     1                   2             Wet or damp   
182122                     1                   1                     Dry   
182123                     1                   2                     Dry   
182124                     3                   1                     Dry   
...                      ...                 ...                     ...   
352694                     1                   1            Frost or ice   
352695                     1                   1             Wet or damp   
352696                     1                   1             Wet or damp   
352697                     2                   2                     Dry   
352702                     1                   1            Frost or ice   

                 Road_Type Urban_or_Rural_Area  Weather_Conditions  \
182120  Single carriageway               Urban  Fine no high winds   
182121  Single carriageway               Urban  Fine no high winds   
182122  Single carriageway               Urban  Fine no high winds   
182123  Single carriageway               Urban  Fine no high winds   
182124  Single carriageway               Urban  Fine no high winds   
...                    ...                 ...                 ...   
352694  Single carriageway               Rural  Fine no high winds   
352695  Single carriageway               Rural  Fine no high winds   
352696  Single carriageway               Rural  Fine no high winds   
352697  Single carriageway               Rural  Fine no high winds   
352702  Single carriageway               Rural  Fine no high winds   

                      Vehicle_Type    Year   Month     Day  DayofWeek  
182120                         Car  2020.0  2020.0  2020.0     2020.0  
182121       Motorcycle over 500cc  2020.0  2020.0  2020.0     2020.0  
182122                         Car  2020.0  2020.0  2020.0     2020.0  
182123                         Car  2020.0  2020.0  2020.0     2020.0  
182124                         Car  2020.0  2020.0  2020.0     2020.0  
...                            ...     ...     ...     ...        ...  
352694                         Car  2020.0  2020.0  2020.0     2020.0  
352695  Motorcycle 125cc and under  2020

<h1>Theres is a 73% total accidents of daylight in 2020 </h1>

In [28]:
df_year20['Light_Conditions'].value_counts().sum()

np.int64(70163)

In [29]:
df_year20['Light_Conditions'].value_counts()

Light_Conditions
Daylight                       51605
Darkness - lights lit          13467
Darkness - no lighting          4085
Darkness - lighting unknown      717
Darkness - lights unlit          289
Name: count, dtype: int64

<h1>Insight no.5</h1>

<h1>Total of 73% Dry road of accident happen in year 2020</h1>

In [30]:
df_year19['Road_Surface_Conditions'].value_counts().sum()

np.int64(71788)

In [31]:
df_year19['Road_Surface_Conditions'].value_counts()

Road_Surface_Conditions
Dry                     52723
Wet or damp             17850
Frost or ice              845
Snow                      268
Flood over 3cm. deep      102
Name: count, dtype: int64

In [32]:
df_year22 = accident[accident['Year'] == 2022]

df_year22

<h1>Insigh no. 6</h1>

<h1>Total of 86% slight accident severity happen in year 2022</h1>

In [33]:
df_year22['Accident_Severity'].value_counts().sum()

np.int64(56805)

In [34]:
df_year22['Accident_Severity'].value_counts()

Accident_Severity
Slight     48916
Serious     7266
Fatal        623
Name: count, dtype: int64

<h1>Insight no. 7</h1>

<h1>Is the accident happen between year 2019 and 2020 increase </h1>

In [35]:
df_year19['Accident_Severity'].value_counts()

Accident_Severity
Slight     60928
Serious     9823
Fatal       1116
Name: count, dtype: int64

In [36]:
df_year19['Accident_Severity'].value_counts().sum()

np.int64(71867)

In [37]:
df_year20['Accident_Severity'].value_counts().sum()

np.int64(70163)

In [38]:
df_year19['Accident_Severity'].value_counts()

Accident_Severity
Slight     60928
Serious     9823
Fatal       1116
Name: count, dtype: int64

<h1>Insight No. 8</h1>

<h1>What year has the most casualties incident </h1>

In [39]:
df_year19['Number_of_Casualties'].value_counts().sum()

np.int64(71867)

In [40]:
df_year20['Number_of_Casualties'].value_counts().sum()

np.int64(70163)

In [41]:
df_year21['Number_of_Casualties'].value_counts().sum()

np.int64(66172)

In [42]:
df_year22['Number_of_Casualties'].value_counts().sum()

np.int64(56805)

<h1>Insight No. 9</h1>

<h1>What vehicle type has the most accident rate in year 2021</h1>

In [43]:
df_year21['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      51745
Van / Goods 3.5 tonnes mgw or under       3359
Motorcycle over 500cc                     2441
Bus or coach (17 or more pass seats)      1683
Goods 7.5 tonnes mgw and over             1477
Motorcycle 125cc and under                1359
Taxi/Private hire car                     1104
Motorcycle 50cc and under                  784
Motorcycle over 125cc and up to 500cc      737
Other vehicle                              578
Goods over 3.5t. and under 7.5t            541
Agricultural vehicle                       181
Minibus (8 - 16 passenger seats)           161
Pedal cycle                                 20
Ridden horse                                 2
Name: count, dtype: int64

In [44]:
df_year21['Vehicle_Type'].value_counts().sum()

np.int64(66172)

<h1>Total of 78% car accident happen in year 2021</h1>

<h1>Insight No. 10</h1>

In [45]:
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
1       200701BS65737           Serious    2019-07-02  51.495029   
2       200701BS66127           Serious           NaT  51.517715   
3       200701BS66128           Serious           NaT  51.495478   
4       200701BS66837            Slight    2019-09-03  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight           NaT  57.374005   
660675  201091NM01881            Slight           NaT  57.232273   
660676  201091NM01935            Slight           NaT  57.585044   
660677  201091NM01964           Serious           NaT  57.214898   
660678  201091NM02142           Serious           NaT  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type    Year   Month     Day  \
0                                        Car  2019.0  2019.0  2019.0   
1                                        Car  2019.0  2019.0  2019.0   
2                      Taxi/Private hire car     NaN     NaN     NaN   
3       Bus or coach (17 or more pass seats)     NaN     NaN     NaN   
4                              Other vehicle  2019.0  2019.0  2019.0   
...                                      ...     ...     ...     ...   
660674          

<h1>Insight 11.</h1>

<h1>What type of vehicle has the most serious accident happen during daylight</h1>

In [46]:
accident_lconditions = accident[(accident['Light_Conditions'] == 'Daylight') & (accident['Accident_Severity'] == 'Serious')]

In [47]:
accident_lconditions

Index Accident_Severity Accident Date   Latitude  \
1       200701BS65737           Serious    2019-07-02  51.495029   
3       200701BS66128           Serious           NaT  51.495478   
5       200701BS67159           Serious           NaT  51.497750   
6       200701BS67207           Serious    2019-09-05  51.501405   
12      200701BS67773           Serious    2019-11-12  51.500823   
...               ...               ...           ...        ...   
660619  201091NK04839           Serious           NaT  58.977273   
660629  201091NL00686           Serious    2022-02-11  57.687933   
660645  201091NL05842           Serious           NaT  60.162660   
660672  201091NM01085           Serious           NaT  57.170725   
660678  201091NM02142           Serious           NaT  57.575210   

       Light_Conditions           District Area  Longitude  \
1              Daylight  Kensington and Chelsea  -0.173647   
3              Daylight  Kensington and Chelsea  -0.202731   
5              Daylight  Kensington and Chelsea  -0.192556   
6              Daylight  Kensington and Chelsea  -0.161577   
12             Daylight  Kensington and Chelsea  -0.193587   
...                 ...                     ...        ...   
660619         Daylight          Orkney Islands  -2.963443   
660629         Daylight        Shetland Islands  -4.176860   
660645         Daylight        Shetland Islands  -1.165232   
660672         Daylight                Highland  -3.852421   
660678         Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
1                          1                   2             Wet or damp   
3                          1                   4                     Dry   
5                          2                   3                     Dry   
6                          1                   2                     Dry   
12                         1                   1                     Dry   
...                      ...                 ...                     ...   
660619                     1                   2                     Dry   
660629                     1                   2                     Dry   
660645                     1                   1                     Dry   
660672                     1                   2                    Snow   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
1       Single carriageway               Urban  Raining no high winds   
3       Single carriageway               Urban     Fine no high winds   
5       Single carriageway               Urban     Fine no high winds   
6         Dual carriageway               Urban     Fine no high winds   
12      Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
660619  Single carriageway               Rural     Fine no high winds   
660629  Single carriageway               Rural      Fine + high winds   
660645  Single carriageway               Rural      Fine + high winds   
660672  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type    Year   Month     Day  \
1                                        Car  2019.0  2019.0  2019.0   
3       Bus or coach (17 or more pass seats)     NaN     NaN     NaN   
5                                        Car     NaN     NaN     NaN   
6        Van / Goods 3.5 tonnes mgw or under  2019.0  2019.0  2019.0   
12                Motorcycle 125cc and under  2019.0  2019.0  2019.0   
...                                      ...     ...     ...     ...   
660619                                   Car     NaN     NaN     NaN   
660629                                   Car  2022.0  2022.0  2022.0   
660645          

<h1>Car has the most serious accident happen in daylight with the total of 75% , which means that the car is more vulnerable in accidents</h1>

In [48]:
accident_lconditions['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      45757
Van / Goods 3.5 tonnes mgw or under       3141
Motorcycle over 500cc                     2418
Bus or coach (17 or more pass seats)      2309
Goods 7.5 tonnes mgw and over             1626
Motorcycle 125cc and under                1379
Taxi/Private hire car                     1218
Motorcycle over 125cc and up to 500cc      691
Motorcycle 50cc and under                  689
Goods over 3.5t. and under 7.5t            593
Other vehicle                              528
Minibus (8 - 16 passenger seats)           191
Agricultural vehicle                       189
Pedal cycle                                 30
Name: count, dtype: int64

In [49]:
accident_lconditions['Vehicle_Type'].value_counts().sum()

np.int64(60759)

<h1>Insight 12</h1>

<h1>In What district Area in Urban has the Serious accident rate happen </h1>

In [50]:
accident_Urban= accident[(accident_lconditions['Urban_or_Rural_Area'] == 'Urban') & (accident['Accident_Severity'] == 'Serious')]

In [51]:
accident_Urban

Index Accident_Severity Accident Date   Latitude  \
1       200701BS65737           Serious    2019-07-02  51.495029   
3       200701BS66128           Serious           NaT  51.495478   
5       200701BS67159           Serious           NaT  51.497750   
6       200701BS67207           Serious    2019-09-05  51.501405   
12      200701BS67773           Serious    2019-11-12  51.500823   
...               ...               ...           ...        ...   
659920  201063CP30410           Serious           NaT  51.712328   
660005  201063DP04110           Serious           NaT  51.766557   
660010  201063DP04610           Serious           NaT  52.514121   
660104  201063DP14210           Serious           NaT  51.767905   
660182  201063DP22110           Serious           NaT  52.505760   

       Light_Conditions           District Area  Longitude  \
1              Daylight  Kensington and Chelsea  -0.173647   
3              Daylight  Kensington and Chelsea  -0.202731   
5              Daylight  Kensington and Chelsea  -0.192556   
6              Daylight  Kensington and Chelsea  -0.161577   
12             Daylight  Kensington and Chelsea  -0.193587   
...                 ...                     ...        ...   
659920         Daylight           Pembrokeshire  -5.036323   
660005         Daylight                   Powys  -3.770561   
660010         Daylight                   Powys  -3.316199   
660104         Daylight                   Powys  -3.776556   
660182         Daylight                   Powys  -3.324201   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
1                          1                   2             Wet or damp   
3                          1                   4                     Dry   
5                          2                   3                     Dry   
6                          1                   2                     Dry   
12                         1                   1                     Dry   
...                      ...                 ...                     ...   
659920                     1                   2                     Dry   
660005                     1                   2                     Dry   
660010                     1                   1                     Dry   
660104                     1                   3                     Dry   
660182                     1                   1                     Dry   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
1       Single carriageway               Urban  Raining no high winds   
3       Single carriageway               Urban     Fine no high winds   
5       Single carriageway               Urban     Fine no high winds   
6         Dual carriageway               Urban     Fine no high winds   
12      Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
659920  Single carriageway               Urban     Fine no high winds   
660005  Single carriageway               Urban     Fine no high winds   
660010  Single carriageway               Urban     Fine no high winds   
660104  Single carriageway               Urban     Fine no high winds   
660182  Single carriageway               Urban     Fine no high winds   

                                Vehicle_Type    Year   Month     Day  \
1                                        Car  2019.0  2019.0  2019.0   
3       Bus or coach (17 or more pass seats)     NaN     NaN     NaN   
5                                        Car     NaN     NaN     NaN   
6        Van / Goods 3.5 tonnes mgw or under  2019.0  2019.0  2019.0   
12                Motorcycle 125cc and under  2019.0  2019.0  2019.0   
...                                      ...     ...     ...     ...   
659920                                   Car     NaN     NaN     NaN   
660005                 Motorcycle over 500cc     NaN     NaN     NaN   
660010          

<h1>Birmingham District Area in Urban has the most seirous happen accident rate with the total of 3% accident rate </h1>

In [52]:
accident_Urban['District Area'].value_counts()

District Area
Birmingham          971
Westminster         614
Leeds               571
Sheffield           467
Liverpool           463
                   ... 
North Norfolk         3
North Shropshire      3
Caradon               3
Castle Morpeth        3
North Cornwall        1
Name: count, Length: 413, dtype: int64

In [53]:
accident_Urban['District Area'].value_counts().sum()

np.int64(34627)

<h1>Insight 13</h1>

<h1>What type of road has the most serious accident rate during daylight</h1>

In [54]:
accident_lconditions = accident[(accident['Light_Conditions'] == 'Daylight') & (accident['Accident_Severity'] == 'Serious')]

In [55]:
accident_lconditions

Index Accident_Severity Accident Date   Latitude  \
1       200701BS65737           Serious    2019-07-02  51.495029   
3       200701BS66128           Serious           NaT  51.495478   
5       200701BS67159           Serious           NaT  51.497750   
6       200701BS67207           Serious    2019-09-05  51.501405   
12      200701BS67773           Serious    2019-11-12  51.500823   
...               ...               ...           ...        ...   
660619  201091NK04839           Serious           NaT  58.977273   
660629  201091NL00686           Serious    2022-02-11  57.687933   
660645  201091NL05842           Serious           NaT  60.162660   
660672  201091NM01085           Serious           NaT  57.170725   
660678  201091NM02142           Serious           NaT  57.575210   

       Light_Conditions           District Area  Longitude  \
1              Daylight  Kensington and Chelsea  -0.173647   
3              Daylight  Kensington and Chelsea  -0.202731   
5              Daylight  Kensington and Chelsea  -0.192556   
6              Daylight  Kensington and Chelsea  -0.161577   
12             Daylight  Kensington and Chelsea  -0.193587   
...                 ...                     ...        ...   
660619         Daylight          Orkney Islands  -2.963443   
660629         Daylight        Shetland Islands  -4.176860   
660645         Daylight        Shetland Islands  -1.165232   
660672         Daylight                Highland  -3.852421   
660678         Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
1                          1                   2             Wet or damp   
3                          1                   4                     Dry   
5                          2                   3                     Dry   
6                          1                   2                     Dry   
12                         1                   1                     Dry   
...                      ...                 ...                     ...   
660619                     1                   2                     Dry   
660629                     1                   2                     Dry   
660645                     1                   1                     Dry   
660672                     1                   2                    Snow   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
1       Single carriageway               Urban  Raining no high winds   
3       Single carriageway               Urban     Fine no high winds   
5       Single carriageway               Urban     Fine no high winds   
6         Dual carriageway               Urban     Fine no high winds   
12      Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
660619  Single carriageway               Rural     Fine no high winds   
660629  Single carriageway               Rural      Fine + high winds   
660645  Single carriageway               Rural      Fine + high winds   
660672  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type    Year   Month     Day  \
1                                        Car  2019.0  2019.0  2019.0   
3       Bus or coach (17 or more pass seats)     NaN     NaN     NaN   
5                                        Car     NaN     NaN     NaN   
6        Van / Goods 3.5 tonnes mgw or under  2019.0  2019.0  2019.0   
12                Motorcycle 125cc and under  2019.0  2019.0  2019.0   
...                                      ...     ...     ...     ...   
660619                                   Car     NaN     NaN     NaN   
660629                                   Car  2022.0  2022.0  2022.0   
660645          

<h1>Total of 80% serious accident happen in daylight in Single carriageway road type , that the Single carriageway most serious accident happen during daylight conditions  </h1>

In [56]:
accident_lconditions["Road_Type"].value_counts()

Road_Type
Single carriageway    48607
Dual carriageway       7619
Roundabout             2592
One way street         1169
Slip road               423
Name: count, dtype: int64

In [57]:
accident_lconditions["Road_Type"].value_counts().sum()

np.int64(60410)

<h1>Insight 14</h1>

<h1>How many serious accidents happen in Urban Area during daylight</h1>

<h1>Total of 34627 serious accidents happen in urban Area during daylight</h1>

In [58]:
accident_Urban= accident[(accident_lconditions['Urban_or_Rural_Area'] == 'Urban') & (accident['Accident_Severity'] == 'Serious')]

In [59]:
accident_Urban["Accident_Severity"].value_counts()

Accident_Severity
Serious    34627
Name: count, dtype: int64

<h1>Insight 15</h1>

<h1>How many serious accidents happen in  Rural Area during daylight</h1>

<h1>Total of 26130 serious accidents happen in urban Area during daylight</h1>

In [60]:
accident_Rural= accident[(accident_lconditions['Urban_or_Rural_Area'] == 'Rural') & (accident['Accident_Severity'] == 'Serious')]

In [61]:
accident_Rural["Accident_Severity"].value_counts()

Accident_Severity
Serious    26130
Name: count, dtype: int64

<h1>Insight 16</h1>

<h1>In What district Area in Rural has the Serious accident rate happen </h1>

In [62]:
accident_Rural= accident[(accident_lconditions['Urban_or_Rural_Area'] == 'Rural') & (accident['Accident_Severity'] == 'Serious')]

In [63]:
accident_Rural

Index Accident_Severity Accident Date   Latitude  \
7185    200701KD50048           Serious           NaT  51.611975   
7197    200701KD50062           Serious           NaT  51.575568   
7216    200701KD50083           Serious           NaT  51.573878   
7225    200701KD50092           Serious           NaT  51.544207   
7363    200701KD50249           Serious    2019-05-07  51.532003   
...               ...               ...           ...        ...   
660619  201091NK04839           Serious           NaT  58.977273   
660629  201091NL00686           Serious    2022-02-11  57.687933   
660645  201091NL05842           Serious           NaT  60.162660   
660672  201091NM01085           Serious           NaT  57.170725   
660678  201091NM02142           Serious           NaT  57.575210   

       Light_Conditions     District Area  Longitude  Number_of_Casualties  \
7185           Daylight          Havering   0.258006                     1   
7197           Daylight          Havering   0.271211                     2   
7216           Daylight          Havering   0.274879                     2   
7225           Daylight          Havering   0.293014                     1   
7363           Daylight          Havering   0.249864                     1   
...                 ...               ...        ...                   ...   
660619         Daylight    Orkney Islands  -2.963443                     1   
660629         Daylight  Shetland Islands  -4.176860                     1   
660645         Daylight  Shetland Islands  -1.165232                     1   
660672         Daylight          Highland  -3.852421                     1   
660678         Daylight          Highland  -3.895673                     1   

        Number_of_Vehicles Road_Surface_Conditions           Road_Type  \
7185                     2                     Dry    Dual carriageway   
7197                     2             Wet or damp    Dual carriageway   
7216                     4             Wet or damp    Dual carriageway   
7225                     2                     Dry  Single carriageway   
7363                     1                     Dry  Single carriageway   
...                    ...                     ...                 ...   
660619                   2                     Dry  Single carriageway   
660629                   2                     Dry  Single carriageway   
660645                   1                     Dry  Single carriageway   
660672                   2                    Snow  Single carriageway   
660678                   1             Wet or damp    Dual carriageway   

       Urban_or_Rural_Area     Weather_Conditions  \
7185                 Rural     Fine no high winds   
7197                 Rural  Raining no high winds   
7216                 Rural  Raining no high winds   
7225                 Rural     Fine no high winds   
7363                 Rural     Fine no high winds   
...                    ...                    ...   
660619               Rural     Fine no high winds   
660629               Rural      Fine + high winds   
660645               Rural      Fine + high winds   
660672               Rural     Fine no high winds   
660678               Rural  Snowing no high winds   

                                Vehicle_Type    Year   Month     Day  \
7185                                     Car     NaN     NaN     NaN   
7197                                     Car     NaN     NaN     NaN   
7216                                     Car     NaN     NaN     NaN   
7225                                     Car     NaN     NaN     NaN   
7363    Bus or coach (17 or more pass seats)  2019.0  2019.0  2019.0   
...                                      ...     ...     ...     ...   
660619                                   Car     NaN     NaN     NaN   
660629                                   Car  2022.0  2022.0  2022.0   
660645                                   Car     NaN     NaN     NaN   
660672          

<h1>East Riding of Yorkshire district area in Rural has the most seious accident happen during dayligh in the total of 398 accidents happen</h1>

In [64]:
accident_Rural['District Area'].value_counts()

District Area
East Riding of Yorkshire       398
Aberdeenshire                  304
Powys                          256
Highland                       254
Wealden                        228
                              ... 
Norwich                          1
Sutton                           1
Newham                           1
Kingston upon Hull, City of      1
Barking and Dagenham             1
Name: count, Length: 400, dtype: int64

In [65]:
accident_Rural['District Area'].value_counts().sum()

np.int64(26130)

<h1>Insight 17</h1>

<h1>What type of road has the serious accident happpen during daylight</h1>

In [66]:
accident_lconditions = accident[(accident['Light_Conditions'] == 'Daylight') & (accident['Accident_Severity'] == 'Serious')]

In [67]:
accident_lconditions

Index Accident_Severity Accident Date   Latitude  \
1       200701BS65737           Serious    2019-07-02  51.495029   
3       200701BS66128           Serious           NaT  51.495478   
5       200701BS67159           Serious           NaT  51.497750   
6       200701BS67207           Serious    2019-09-05  51.501405   
12      200701BS67773           Serious    2019-11-12  51.500823   
...               ...               ...           ...        ...   
660619  201091NK04839           Serious           NaT  58.977273   
660629  201091NL00686           Serious    2022-02-11  57.687933   
660645  201091NL05842           Serious           NaT  60.162660   
660672  201091NM01085           Serious           NaT  57.170725   
660678  201091NM02142           Serious           NaT  57.575210   

       Light_Conditions           District Area  Longitude  \
1              Daylight  Kensington and Chelsea  -0.173647   
3              Daylight  Kensington and Chelsea  -0.202731   
5              Daylight  Kensington and Chelsea  -0.192556   
6              Daylight  Kensington and Chelsea  -0.161577   
12             Daylight  Kensington and Chelsea  -0.193587   
...                 ...                     ...        ...   
660619         Daylight          Orkney Islands  -2.963443   
660629         Daylight        Shetland Islands  -4.176860   
660645         Daylight        Shetland Islands  -1.165232   
660672         Daylight                Highland  -3.852421   
660678         Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
1                          1                   2             Wet or damp   
3                          1                   4                     Dry   
5                          2                   3                     Dry   
6                          1                   2                     Dry   
12                         1                   1                     Dry   
...                      ...                 ...                     ...   
660619                     1                   2                     Dry   
660629                     1                   2                     Dry   
660645                     1                   1                     Dry   
660672                     1                   2                    Snow   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
1       Single carriageway               Urban  Raining no high winds   
3       Single carriageway               Urban     Fine no high winds   
5       Single carriageway               Urban     Fine no high winds   
6         Dual carriageway               Urban     Fine no high winds   
12      Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
660619  Single carriageway               Rural     Fine no high winds   
660629  Single carriageway               Rural      Fine + high winds   
660645  Single carriageway               Rural      Fine + high winds   
660672  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type    Year   Month     Day  \
1                                        Car  2019.0  2019.0  2019.0   
3       Bus or coach (17 or more pass seats)     NaN     NaN     NaN   
5                                        Car     NaN     NaN     NaN   
6        Van / Goods 3.5 tonnes mgw or under  2019.0  2019.0  2019.0   
12                Motorcycle 125cc and under  2019.0  2019.0  2019.0   
...                                      ...     ...     ...     ...   
660619                                   Car     NaN     NaN     NaN   
660629                                   Car  2022.0  2022.0  2022.0   
660645          

<h1>Total of 76% of serious accident rate happen in Dry road surface conditions during daylight. The most prone in accident is dry surface during daylight</h1>

In [68]:
accident_lconditions["Road_Surface_Conditions"].value_counts()

Road_Surface_Conditions
Dry                     46535
Wet or damp             12829
Frost or ice              985
Snow                      298
Flood over 3cm. deep       73
Name: count, dtype: int64

In [69]:
accident_lconditions["Road_Surface_Conditions"].value_counts().sum()

np.int64(60720)

<h1>Insight 18</h1>

<h1>How do light conditions, casualty numbers, and the type of area (urban or rural) impact the occurrence and distribution of accidents?</h1>

In [70]:
accident_lconditions= np.round(accident.groupby(['Light_Conditions', 'Number_of_Casualties', 'Weather_Conditions'])['Urban_or_Rural_Area'].size(), 1)

In [71]:
accident_lconditions.unstack()

Weather_Conditions                                Fine + high winds  \
Light_Conditions            Number_of_Casualties                      
Darkness - lighting unknown 1                                  43.0   
                            2                                  10.0   
                            3                                   5.0   
                            4                                   1.0   
                            5                                   NaN   
...                                                             ...   
Daylight                    42                                  NaN   
                            43                                  NaN   
                            45                                  NaN   
                            47                                  NaN   
                            62                                  NaN   

Weather_Conditions                                Fine no high winds  \
Light_Conditions            Number_of_Casualties                       
Darkness - lighting unknown 1                                 2862.0   
                            2                                  566.0   
                            3                                  156.0   
                            4                                   70.0   
                            5                                   18.0   
...                                                              ...   
Daylight                    42                                   1.0   
                            43                                   1.0   
                            45                                   1.0   
                            47                                   NaN   
                            62                                   1.0   

Weather_Conditions                                Fog or mist  Other  \
Light_Conditions            Number_of_Casualties                       
Darkness - lighting unknown 1                            48.0  151.0   
                            2                            12.0   25.0   
                            3                             3.0    8.0   
                            4                             2.0    1.0   
                            5                             NaN    1.0   
...                                                       ...    ...   
Daylight                    42                            1.0    NaN   
                            43                            NaN    NaN   
                            45                            NaN    NaN   
                            47                            NaN    NaN   
                            62                            NaN    NaN   

Weather_Conditions                                Raining + high winds  \
Light_Conditions            Number_of_Casualties                         
Darkness - lighting unknown 1                                     82.0   
                            2                                     26.0   
                            3                                      2.0   
                            4                                      NaN   
                            5                                      NaN   
...                                                                ...   
Daylight                    42                                     NaN   
                            43                                     NaN   
                            45                                     NaN   
                            47                                     NaN   
                            62                                     NaN   

Weather_Conditions                                Raining no high winds  \
Light_Conditions            Number_of_Casualties                          
Darkness - lighting unknown 1                                     498.0   
              

<h1>If specific rural areas see more accidents in low light conditions, it could point to regional infrastructure problems, like insufficient street lighting or poorly maintained roads. Conversely, a high number of casualties in urban areas with good lighting may suggest that factors such as high population density, crowded intersections, or traffic flow patterns play a larger role in the severity of accidents.</h1>

<h1>Insight 19</h1>

<h1>What role do light conditions, casualty rates, and district type (urban or rural) play in shaping the distribution of accidents?</h1>

In [72]:
wehther_conditions = np.round(accident.groupby(['Light_Conditions', 'Number_of_Casualties', 'District Area'])['Urban_or_Rural_Area'].size(), 1)

In [73]:
wehther_conditions.unstack()

District Area                                     Aberdeen City  \
Light_Conditions            Number_of_Casualties                  
Darkness - lighting unknown 1                               8.0   
                            2                               NaN   
                            3                               NaN   
                            4                               NaN   
                            5                               NaN   
...                                                         ...   
Daylight                    42                              NaN   
                            43                              NaN   
                            45                              NaN   
                            47                              NaN   
                            62                              NaN   

District Area                                     Aberdeenshire  Adur  \
Light_Conditions            Number_of_Casualties                        
Darkness - lighting unknown 1                               3.0   3.0   
                            2                               NaN   1.0   
                            3                               NaN   NaN   
                            4                               NaN   NaN   
                            5                               NaN   NaN   
...                                                         ...   ...   
Daylight                    42                              NaN   NaN   
                            43                              NaN   NaN   
                            45                              NaN   NaN   
                            47                              NaN   NaN   
                            62                              NaN   NaN   

District Area                                     Allerdale  Alnwick  \
Light_Conditions            Number_of_Casualties                       
Darkness - lighting unknown 1                           6.0      1.0   
                            2                           NaN      NaN   
                            3                           NaN      NaN   
                            4                           NaN      NaN   
                            5                           NaN      NaN   
...                                                     ...      ...   
Daylight                    42                          NaN      NaN   
                            43                          NaN      NaN   
                            45                          NaN      NaN   
                            47                          NaN      NaN   
                            62                          NaN      NaN   

District Area                                     Amber Valley  Angus  \
Light_Conditions            Number_of_Casualties                        
Darkness - lighting unknown 1                             25.0    2.0   
                            2                              2.0    NaN   
                            3                              1.0    NaN   
                            4                              1.0    NaN   
                            5                              NaN    NaN   
...                                                        ...    ...   
Daylight                    42                             NaN    NaN   
                            43                             NaN    NaN   
                            45                             NaN    NaN   
                            47                             NaN    NaN   
                            62                             NaN    NaN   

District Area                                     Argyll and Bute  Arun  \
Light_Conditions            Number_of_Casualties                          
Darkness - lighting unknown 1                                 NaN  12.0   
                            2                                 1.0 

<h1>If specific rural areas have a higher number of accidents under low light conditions, it could point to regional infrastructure problems, such as insufficient street lighting or poorly maintained roads. In contrast, a significant number of casualties in urban areas with adequate lighting may suggest that factors like high population density, congested intersections, or traffic patterns play a larger role in the severity of accidents.</h1>

<h1>Insight 20</h1>

<h1>How do lighting, casualty rates, and road types affect the occurrence and distribution of accidents in urban and rural areas?</h1>


In [74]:
weather_conditions = np.round(accident.groupby(['Light_Conditions', 'Number_of_Casualties', 'Road_Type'])['Urban_or_Rural_Area'].size(), 1)

In [75]:
weather_conditions.unstack()

Road_Type                                         Dual carriageway  \
Light_Conditions            Number_of_Casualties                     
Darkness - lighting unknown 1                                684.0   
                            2                                143.0   
                            3                                 44.0   
                            4                                 22.0   
                            5                                  5.0   
...                                                            ...   
Daylight                    42                                 1.0   
                            43                                 1.0   
                            45                                 1.0   
                            47                                 1.0   
                            62                                 NaN   

Road_Type                                         One way street  Roundabout  \
Light_Conditions            Number_of_Casualties                               
Darkness - lighting unknown 1                              104.0       378.0   
                            2                               17.0        70.0   
                            3                                2.0        17.0   
                            4                                1.0         4.0   
                            5                                NaN         1.0   
...                                                          ...         ...   
Daylight                    42                               NaN         NaN   
                            43                               NaN         NaN   
                            45                               NaN         NaN   
                            47                               NaN         NaN   
                            62                               NaN         NaN   

Road_Type                                         Single carriageway  \
Light_Conditions            Number_of_Casualties                       
Darkness - lighting unknown 1                                 3707.0   
                            2                                  680.0   
                            3                                  178.0   
                            4                                   68.0   
                            5                                   22.0   
...                                                              ...   
Daylight                    42                                   1.0   
                            43                                   NaN   
                            45                                   NaN   
                            47                                   NaN   
                            62                                   1.0   

Road_Type                                         Slip road  
Light_Conditions            Number_of_Casualties             
Darkness - lighting unknown 1                          64.0  
                            2                          16.0  
                            3                           4.0  
                            4                           2.0  
                            5                           NaN  
...                                                     ...  
Daylight                    42                          NaN  
                            43                          NaN  
                            45                          NaN  
                            47                          NaN  
                            62                          NaN  

[88 rows x 5 columns]

<h1>If accidents on high-speed roads (like motorways) are more common in rural areas under low light conditions, it could highlight the need for enhanced visibility measures, such as better road lighting or reflective markings. In contrast, a higher number of casualties on urban roads with good lighting may indicate that factors such as heavy traffic, intersections, or pedestrian activity contribute more to the severity of accidents.</h1>